### PR TITLE
Adds the ExtendedSourceFileDto object

### PR DESCRIPTION
**Pull request by phinpho**
To represent the current response of the `getJobFiles` endpoint, the object to represent the items has been changed to allow the usage of the following properties:

- `localeIds`: An array of the locales assigned for the files;
- `fileUid`: The Uid of the file, also present in the `getJob` response;
- `name`: The name of the file, also present in the `getJob` response.

The property `localeIds` can also be undefined, because the API docs are currently missing this value, the interface assumes it can be present or not.

### CODE REVIEW

File: api/jobs/dto/extended-source-file-dto.ts
Comment (dimitrystd) [inline api/jobs/dto/extended-source-file-dto.ts:3]: We already have a mess with `SourceFileDto` vs `FullSourceFileDto` (I didn't check how we use them). Adding one more magic name `ExtendedSourceFileDto` will confuse everyone. I would recommend making it explicit like `JobFileDto`, or something like this.

File: api/jobs/dto/full-source-file-dto.ts
Comment (dimitrystd) [inline api/jobs/dto/full-source-file-dto.ts:3]: @PavelLoparev  I would recommend checking where we use it. Maybe we can eliminate it. I hope jobs API is consistent and we have 3 fields everywhere (uri, name, fileUid).

**Review by dimitrystd — CHANGES_REQUESTED** [review]

File: api/jobs/dto/full-source-file-dto.ts
Comment (dimitrystd) [reply api/jobs/dto/full-source-file-dto.ts:3]: Thx for your contribution. Let's wait for another review from Smartling dev.

Comment (PavelLoparev) [reply api/jobs/dto/full-source-file-dto.ts:3]: @dimitrystd unfortunately jobs api returns different file dtos from different endpoints. that's why we have `SourceFileDto` and `FullSourceFileDto` (at least it's true for the time these dtos were introduced in this sdk, I didn't check jobs api recently).

Comment (PavelLoparev) [reply api/jobs/dto/full-source-file-dto.ts:3]: Ok, https://api-reference.smartling.com/#tag/Jobs is a bit outdated on responses dto. I'll notify jobs api team about that.

**Review by PavelLoparev — COMMENTED** [review]
Hi @phinpho,

> The property localeIds can also be undefined, because the API docs are currently missing this value, the interface assumes it can be present or not.

Seems like the api reference doc is a bit outdated on that endpoint. I believe response for `getJobFiles` call should include `localeIds` in every source file object, I believe it's not necessary to mark it optional (at least tried manually and got this property for every file in the response). But it can be an empty array though.

Other than that PR looks good to me.

**Comment by PavelLoparev** [comment]
> > Hi @phinpho,
> > > The property localeIds can also be undefined, because the API docs are currently missing this value, the interface assumes it can be present or not.
> > 
> > 
> > Seems like the api reference doc is a bit outdated on that endpoint. I believe response for `getJobFiles` call should include `localeIds` in every source file object, I believe it's not necessary to mark it optional (at least tried manually and got this property for every file in the response). But it can be an empty array though.
> > Other than that PR looks good to me.
> 
> So, in case the response for `getJobFiles` can return the `localeIds` always, I think it isn't up to me to define that, I'm adding this change based on what I'm seeing on the API response, but because I cannot be sure about the response structure, I believe that Smartling devs should set the property `localeIds` as not optional for `FullSourceFileDto` object.
> 
> Can you consider adding this change right after this PR?

Sure. Released 1.5.1

File: api/jobs/dto/full-source-file-dto.ts
Comment (dimitrystd) [reply api/jobs/dto/full-source-file-dto.ts:3]: @PavelLoparev I contacted them already

**Pull request merged**
